src/main/java/com/seailz/discordjar/model/webhook/Webhook.java

```diff
@@ -0,0 +1,103 @@
+ */
+public record Webhook(
+        Snowflake id,
+        int type,
+        Snowflake guildId,
+        Snowflake channelId,
+        User user,
+        String name,
+        String avatar,
+        String token,
+        Snowflake applicationId,
+        Guild sourceGuild,
+        Channel sourceChannel,
+        String url,
+        DiscordJar discordJar
+) implements Compilerable {
+    @Override
+    public JSONObject compile() {
+        return new JSONObject()
+                .put("id", id.id())
+                .put("type", type)
+                .put("guild_id", guildId.id())
+                .put("channel_id", channelId.id())
+                .put("user", user.compile())
+                .put("name", name)
+                .put("avatar", avatar)
+                .put("token", token)
+                .put("application_id", applicationId.id())
+                .put("source_guild", sourceGuild.compile())
+                .put("source_channel", sourceChannel.compile())
+                .put("url", url);
+    }
+
+    public static Webhook decompile(JSONObject json, DiscordJar discordJar) {
+        return new Webhook(
+                json.has("id") ? () -> json.getString("id") : null,
+                json.has("type") ? json.getInt("type") : 0,
+                json.has("guild_id") ? () -> json.getString("guild_id") : null,
+                json.has("channel_id") ? () -> json.getString("channel_id") : null,
+                json.has("user") ? User.decompile(json.getJSONObject("user"), discordJar) : null,
+                json.has("name") ? json.getString("name") : null,
+                json.has("avatar") ? json.getString("avatar") : null,
+                json.has("token") ? json.getString("token") : null,
+                json.has("application_id") ? () -> json.getString("application_id") : null,
+                json.has("source_guild") ? Guild.decompile(json.getJSONObject("source_guild"), discordJar) : null,
+                json.has("source_channel") ? Channel.decompile(json.getJSONObject("source_channel"), discordJar) : null,
+                json.has("url") ? json.getString("url") : null,
+                discordJar
+        );
+    }
+
+    /**
+     * @return This Webhook as an {@link com.seailz.discordjar.model.webhook.IncomingWebhook}.
+     */
+    public IncomingWebhook asIncomingWebhook() {
+        return new IncomingWebhook(name, type, channelId, token, avatar, guildId, id, applicationId, user, discordJar);
+    }
+
+    /**
+     * Deletes this Webhook.
+     * @throws DiscordRequest.UnhandledDiscordAPIErrorException Thrown when an unexpected error is returned from the Discord API.
+     */
+    public void delete() throws DiscordRequest.UnhandledDiscordAPIErrorException {
+        DiscordResponse response = new DiscordRequest(
+                new JSONObject(),
+                new HashMap<>(),
+                URLS.DELETE.CHANNEL.DELETE_WEBHOOK.replace("{webhook.id}", id.id()).replace("{webhook.token}", token),
```

Comment:
If the token is null or an empty string, this should attempt just using the bot token.

Reply:
As far as I know, Webhook tokens are unique to the Webhook so that would never work, i will probably just implement a null-or-empty check here.

Reply:
This has been implemented